tests/controlboard/ControlBoardPositionControlTest.cc

```diff
@@ -0,0 +1,145 @@
+                int nJointsConfigured{};
+                ASSERT_TRUE(iPositionControl->getAxes(&nJointsConfigured));
+                std::cerr << "Number of joints configured: " << nJointsConfigured << std::endl;
+
+                configured = true;
+                std::cerr << "========== Test configured" << std::endl;
+            });
+    }
+
+    // Get SDF model name from test parameter
+    gz::sim::TestFixture testFixture;
+    std::string deviceScopedName = "model/single_pendulum/controlboard_plugin_device";
+    double linkMass{1};
+    double linkLength{1.0};
+    double linkInertiaAtLinkEnd{0.3352}; // Computed with parallel axis theorem
+    int plannedIterations{1000};
+    int iterations{0};
+    std::vector<double> trackingErrors{};
+    double acceptedTolerance{5e-2};
+    bool configured{false};
+    gz::math::Vector3d gravity;
+    gz::sim::Entity modelEntity;
+    gz::sim::Model model;
+    gz::sim::Joint joint;
+    yarp::dev::PolyDriver* driver;
+    yarp::dev::IPositionControl* iPositionControl = nullptr;
+    yarp::dev::IControlMode* iControlMode = nullptr;
+    yarp::dev::IEncoders* iEncoders = nullptr;
+};
+
+TEST_F(ControlBoardPositionFixture, CheckPositionTrackingWithTrajectoryGenerationUsingPendulumModel)
+{
+    auto refPosition{90.0};
+    bool motionDone{false};
+    double jointPosition, jointPosError;
+
+    testFixture
+        .OnPostUpdate(
+            [&](const gz::sim::UpdateInfo& _info, const gz::sim::EntityComponentManager& _ecm) {
+                // std::cerr << "========== Iteration: " << iterations << std::endl;
+
+                iEncoders->getEncoder(0, &jointPosition);
+                iPositionControl->checkMotionDone(0, &motionDone);
+
+                // std::cerr << "ref position: " << refTrajectory[iterations] << std::endl;
+                // std::cerr << "joint position: " << jointPosition << std::endl;
+
+                iterations++;
+            })
+        .
+        // The moment we finalize, the configure callback is called
+        Finalize();
+
+    int modeSet{};
+    iControlMode->getControlMode(0, &modeSet);
+    ASSERT_TRUE(modeSet == VOCAB_CM_POSITION);
+
+    // Set reference position
+    iPositionControl->positionMove(0, refPosition);
+
+    // Setup simulation server, this will call the post-update callbacks.
+    // It also calls pre-update and update callbacks if those are being used.
+    while (!motionDone)
+    {
+        std::cerr << "Running server" << std::endl;
+        testFixture.Server()->Run(true, plannedIterations, false);
+        jointPosError = abs(refPosition - jointPosition);
+        std::cerr << "Joint position error: " << jointPosError << std::endl;
+    }
+
+    // Final assertions
+    ASSERT_TRUE(configured);
+    ASSERT_TRUE(motionDone);
+
+    // Verify that the final error is within the accepted tolerance
+    ASSERT_LT(jointPosError, acceptedTolerance);
+    std::cerr << "Final tracking error: " << jointPosError << std::endl;
```

Comment:
```suggestion
    std::cerr << "Final tracking error: " << jointPosError << std::endl;
    ASSERT_LT(jointPosError, acceptedTolerance);
```

Not sure if `ASSERT_LT` actually ends the test, but if that is the case it is better to first print the final tracking error, and then check for it.

Reply:
Thank you! yes `ASSERT_LT` terminates the test hence I would not be able to read the error.